File: src/main/scala/ru/sbrf/gg/load/builder/AffinityParticleKeyBuilder.scala
package ru.sbrf.gg.load.builder

import com.sbt.dpl.gridgain.AffinityParticleKey
import ru.sbrf.gg.load.TableInfo

/**
  */
class AffinityParticleKeyBuilder extends ReflectionBuilder {
    override def build(line: Array[String], tableInfo: TableInfo): Any = {
        if (tableInfo.key != classOf[AffinityParticleKey])
            throw new RuntimeException("Not AffinityParticle Key!!!")

        val id = fieldValue(line, tableInfo.id).asInstanceOf[Long]
        val partitionId = fieldValue(line, tableInfo.partitionId).asInstanceOf[Long]
        val rootId = fieldValue(line, tableInfo.rootId).asInstanceOf[Long]

        new AffinityParticleKey(id, partitionId, rootId)
    }
}

File: src/main/java/ru/sbrf/gg/load/builder/AffinityParticleKeyBuilder.java
package ru.sbrf.gg.load.builder;

import com.sbt.DelimetedStringParser;
import com.sbt.dpl.gridgain.AffinityParticleKey;
import ru.sbrf.gg.load.TableInfo;

/**
 */
public class AffinityParticleKeyBuilder extends ReflectionBuilder implements ObjectBuilder {
    @Override public Object build(String line, TableInfo tableInfo) {
        if (tableInfo.key != AffinityParticleKey.class)
            throw new RuntimeException("Not AffinityParticle Key!!!");

        int[] indexes = new int[]{0, 0, line.length()};

        Long id = DelimetedStringParser._long(tableInfo.id.idx, line, indexes);
        Long partitionId = DelimetedStringParser._long(tableInfo.partitionId.idx, line, indexes);
        Long rootId = DelimetedStringParser._long(tableInfo.rootId.idx, line, indexes);

        return new AffinityParticleKey(id, partitionId, rootId);
    }

    @Override public int compare(Object first, Object second) {
        AffinityParticleKey f = (AffinityParticleKey)first;
        AffinityParticleKey s = (AffinityParticleKey)second;

        if (f.getId() != s.getId())
            return f.getId() > s.getId() ? 1 : -1;

        if (f.getPartitionId() != s.getPartitionId())
            return f.getPartitionId() > s.getPartitionId() ? 1 : -1;

        if (f.getRootId() != s.getRootId())
            return f.getRootId() > s.getRootId() ? 1 : -1;

        return 0;
    }
}
